File: go1_control/src/motor_control.cpp
#include "rclcpp/rclcpp.hpp"
#include "ros2_unitree_legged_msgs/msg/motor_cmd.hpp"

class MyMotorControl : public rclcpp::Node
{
public:
    MyMotorControl() : Node("my_motor_control")
    {
        // Create publishers for each joint you want to control
        // (or all 12 if you want full control)
        FR_hip_pub_   = create_publisher<Cmd>("/FR_hip_controller/command", 1);
        FR_thigh_pub_ = create_publisher<Cmd>("/FR_thigh_controller/command", 1);
        FR_calf_pub_  = create_publisher<Cmd>("/FR_calf_controller/command", 1);
        // ... repeat for all 12 joints

        timer_ = create_wall_timer(
            std::chrono::milliseconds(2),  // 500Hz
            std::bind(&MyMotorControl::send_cmd, this));
    }

private:
    using Cmd = ros2_unitree_legged_msgs::msg::MotorCmd;

    void send_cmd()
    {
        Cmd cmd;
        cmd.mode = 0x0A;   // position control
        cmd.kp   = 20.0;   // stiffness - start LOW (5.0) for safety
        cmd.kd   = 0.5;    // damping
        cmd.tau  = 0.0;

        // FR hip to 0 radians
        cmd.q = 0.0;
        FR_hip_pub_->publish(cmd);

        // FR thigh to 0.8 rad (~45 deg, standing pose)
        cmd.q = 0.8;
        FR_thigh_pub_->publish(cmd);

        // FR calf to -1.6 rad
        cmd.q = -1.6;
        FR_calf_pub_->publish(cmd);
    }

    rclcpp::Publisher<Cmd>::SharedPtr FR_hip_pub_, FR_thigh_pub_, FR_calf_pub_;
    rclcpp::TimerBase::SharedPtr timer_;
};

int main(int argc, char **argv)
{
    rclcpp::init(argc, argv);
    rclcpp::spin(std::make_shared<MyMotorControl>());
    rclcpp::shutdown();
}

File: go1_control/go1_control/motor_control.py
import rclpy
from rclpy.node import Node
from ros2_unitree_legged_msgs.msg import MotorCmd

class MyMotorControl(Node):
    def __init__(self):
        super().__init__('my_motor_control')

        topics = [
            'FR_hip', 'FR_thigh', 'FR_calf',
            'FL_hip', 'FL_thigh', 'FL_calf',
            'RR_hip', 'RR_thigh', 'RR_calf',
            'RL_hip', 'RL_thigh', 'RL_calf',
        ]

        self.pubs = {
            name: self.create_publisher(MotorCmd, f'/{name}_controller/command', 1)
            for name in topics
        }

        self.timer = self.create_timer(0.002, self.send_cmd)  # 500 Hz

    def send_cmd(self):
        cmd = MotorCmd()
        cmd.mode = 0x0A  # position control
        cmd.kp   = 20.0  # stiffness - start LOW (5.0) for safety
        cmd.kd   = 0.5   # damping
        cmd.tau  = 0.0

        cmd.q = 0.0
        for name in ('FR_hip', 'FL_hip', 'RR_hip', 'RL_hip'):
            self.pubs[name].publish(cmd)

        cmd.q = 0.8
        for name in ('FR_thigh', 'FL_thigh', 'RR_thigh', 'RL_thigh'):
            self.pubs[name].publish(cmd)

        cmd.q = -1.6
        for name in ('FR_calf', 'FL_calf', 'RR_calf', 'RL_calf'):
            self.pubs[name].publish(cmd)


def main(args=None):
    rclpy.init(args=args)
    node = MyMotorControl()
    rclpy.spin(node)
    rclpy.shutdown()


if __name__ == '__main__':
    main()
